DSharpPlus.Commands/ICommandExecutor.cs

```diff
@@ -0,0 +1,23 @@
+namespace DSharpPlus.Commands;
+
+using System.Threading;
+using System.Threading.Tasks;
+
+using DSharpPlus.Commands.Trees;
+
+public interface ICommandExecutor
+{
+
+    /// <summary>
+    /// Executes a command asynchronously.
+    /// </summary>
+    /// <param name="context">The context of the command.</param>
+    /// <param name="awaitCommandExecution">Whether to not return until the command has finished executing.</param>
+    /// <param name="cancellationToken">The cancellation token to use.</param>
+    public ValueTask ExecuteAsync
```

Comment:
Could this be a delegate instead of an interface

Reply:
it could, having an interface is probably preferable though since it gives DSP.C a concrete instance of a concrete type to work with

Reply:
Future proofing, got it.